### PR TITLE
Update to deployment-pipelines v3

### PR DESCRIPTION
**Pull request by djordjelacmanovic**
## Summary

Reusable image build and deployment actions use `actions/checkout`. Depending on the workflow trigger used this action may unexpectedly checkout default branch HEAD.

**Related issue**: [#1192](https://app.productive.io/1-infinum/tasks/14091509)

### Type

- [ ] **Feature**: This pull request introduces a new feature.
- [x] **Bug fix**: This pull request fixes a bug.
- [ ] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

#### Additional information

- [ ] This pull request introduces a **breaking change**.

### Description

- updated to use `v3.0.0` deployment pipelines which add mandatory `git_ref` input
- resolve correct git revision for the common use case (`workflow_run` trigger) and
  - use it as image tag
  - pass it to deployment actions
- added TODOs and short docs explaining the GHA trigger gotchas

## Checklist

- [x] I have performed a self-review of my own code.
- [ ] I have tested my changes, including edge cases.
- [ ] I have added necessary tests for the changes introduced (if applicable).
- [ ] I have updated the documentation to reflect my changes (if applicable).

### CODE REVIEW

File: templates/SETUP.md [hunk 1]
```diff
@@ -7,6 +7,33 @@ To complete project setup from this template, complete the following steps:
 - [ ] Update `README.md`
 - [ ] Update infrastructure concerns in coordination with the DevOps team
     - [ ] Specify ECS task definitions (`/infra/ecs/*`)
-    - [ ] Update ECS deployment configuration (`.github/workflows/deploy-*`)
+    - [ ] Update ECS deployment configuration (`.github/workflows/deploy-*`).
+    - [ ] Review [action triggers gotchas](#github-actions-triggers).
 - [ ] Review and resolve `TODO` comments
 - [ ] Delete the `SETUP.md` file
+
+#### Github Actions trigger gotchas
```
Comment (vr4b4c) [inline templates/SETUP.md:15]: 🙇

**Review by vr4b4c — APPROVED** [review]
